app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesComposables.kt

```diff
@@ -403,6 +405,11 @@ fun PoweredByPocketHeader(
     val linkStartIndex = text.indexOf(link)
     val linkEndIndex = linkStartIndex + link.length
 
+    val wallpaperState = components.appStore
+        .observeAsComposableState { state -> state.wallpaperState }.value
+
+    val wallpaperAdaptedTextColor = wallpaperState?.currentWallpaper?.textColor?.let { Color(it) }
```

Comment:
the call to `components` will break the previews, unfortunately. the text color will need to be passed-in as a parameter and the default could then just be `FirefoxTheme.colors.textPrimary`. 

As @gabrielluong said, this can be follow-up.

Reply:
Ah, this is an even more compelling reason that goes with my initial thoughts.

Reply:
I fixed the preview by passing the store with a default value. I do agree that when we finish migrating the whole screen to compose passing the colors directly and observing changes in the topmost composable is a cleaner solution.

Reply:
If my count is correct, you would need to do the duplicate code 3-4 times, I think this would still honestly be better than having `appStore` as a parameter to these components.

Reply:
While I might agree with the fact that passing the color is better in some cases, IMO there are more arguments that support this implementation at this stage.

The viewholders do not need the info contained in the state, their UI does not change, with the exception of the section header. Moreover, if we add new sections we will add other code duplications, for components that are temporary in the plan of migrating the screen to full Compose. 

As we migrate the screen to Compose the section headers are still the ones that need to change depending on the state, having the state in the viewholders would add complexity to the refactoring. 

As for the previews, If the section header text is supposed to be wallpaper dependent, I think having the possibility to represent that dependency directly (by adding custom wallpapers as the current wallpaper in the preview param state) is more suggestive than having it change by an explicit color.

Reply:
As for the previews, If the section header text is supposed to be wallpaper dependent, I think having the possibility to represent that dependency directly (by adding custom wallpapers as the current wallpaper in the preview param state) is more suggestive than having it change by an explicit color. 
This is illustrated in the last commit: https://github.com/mozilla-mobile/fenix/pull/26631/commits/1e7eda26b1cd37a7b9e63ca5f564659b4f9a9045.

Having these previews also adds to the argument that the current parent of the section header (the viewholders) does not need to instruct the child on what color to use as they do not need to use that information elsewhere. As a side note, there also is the hypothetical discussion that if we would need 10 values from the state we would pass the whole state, not 10 separate params.

Reply:
Personally, I'm in the camp of simple passed-in parameters. If we do one day make it to 100% Compose, there _will_ be Compose code interwoven with interactors/use cases/stores etc, but the ultimate front-end UI `@Compose` functions should be kept as dumb as possible and only get passed parameters that directly drive the experience (lambdas, specific colors, data, etc.). 

An example of our transition to 100% has been our use of ViewHolders. Looking at `InactiveTabViewHolder`, `SyncedTabsPageViewHolder`, and `RecentBookmarksViewHolder`, the state/store is observed in the ViewHolder, and the Composables they contain only receive what is absolutely necessary. This keeps the API super clean and transparent, and streamlines `@Previews` since they don't need to do any substantial object instantiation. When we transition off of ViewHolders, there will still exist an intermediary layer between the interactors/use cases/stores etc and the front-end Consumers (e.g. InactiveTabsList, RecentBookmarks), so it behooves us to do that work now and keep that separation.